src/Nuget.PackageIndex/NupkgLocalPackageLoader.cs

```diff
@@ -46,103 +46,103 @@ internal class NupkgLocalPackageLoader : ILocalPackageLoader
+                        //      So dlls under contract would have all package's target frameworks, dlls under fx folder would have 
+                        //      only that particular target fx.
+                        //
+                        // New nuget format:
+                        //      - assemblies live under lib\fx folders as before
+                        //      - instdead of contract folder there is now Ref folder which contains common contracts. Unlice old contract
+                        //        folder, Ref folder might contain subfolder Any meaning that those contracts belong to any target fx; and
+                        //        corresponding Fx subfolder, which would contain types specific to particular FX.
+                        // 
+                        //      Logic for new format:
+                        //       - if there is a ref folder use assemblies under ref\any and then ref\fx; ignore lib assemblies
+                        //       - if there no ref folder just use assemblies from lib\fx subfolders
+                        //  
+                        // Note: just in case we process all assemblies lib, ref and contract. Assemblies with ref/contract that are in 
+                        // lib folder would not contain any types normally, so they shoudl be quick to process.
+
+                        var packageTargetFrameworks = package.GetSupportedFrameworks().Select(x => VersionUtility.GetShortFrameworkName(x));
+                        var assembliesMetadata = new List<AssemblyMetadata>();
+                        foreach (var assembly in allAssemblies)
                         {
-                            continue;
-                        }
+                            var assemblyFullPath = Path.Combine(packageFolder, assembly.Path);
+                            if (!File.Exists(assemblyFullPath))
+                            {
+                                assemblyFullPath = Path.Combine(packageFolder, Path.GetFileNameWithoutExtension(nupkgFilePath), assembly.Path);
+                            }
 
-                        IEnumerable<string> assemblyTargetFrameworks =  assembly.SupportedFrameworks.Select(x => VersionUtility.GetShortFrameworkName(x));
-                        if (!assemblyTargetFrameworks.Any())
-                        {
-                            assemblyTargetFrameworks = packageTargetFrameworks;
-                        }
+                            if (!File.Exists(assemblyFullPath))
+                            {
+                                continue;
+                            }
 
-                        assembliesMetadata.Add(
-                            new AssemblyMetadata
+                            IEnumerable<string> assemblyTargetFrameworks =  assembly.SupportedFrameworks.Select(x => VersionUtility.GetShortFrameworkName(x));
```

Comment:
Right now the format is a little bit in flux, but once it stabilizes, it could definitely have a hard dependency.

@davidfowl any thoughts on the urgency of this?